pycrostates/segmentation/segmentation.py

```diff
@@ -147,9 +145,187 @@ def compute_parameters(
             return_dist=return_dist,
         )
 
+    def compute_transition_matrix(self, stat="probability", ignore_self=True):
+        """Compute the observed transition matrix.
+
+        Count the number of transitions from one state to another
+        and aggregate the result as statistic.
+        Transition "from" and "to" unlabeled segments ``-1`` are ignored.
+
+        Parameters
+        ----------
+        stat : str
+            Aggregate statistic to compute transitions. Can be:
+
+            * ``count``: show the number of observations of each transition.
+            * ``probability`` or ``proportion``: normalize count such as the
+              probabilities along the first axis is always equal to ``1``.
+            * ``percent``: normalize count such as the
+              probabilities along the first axis is always equal to ``100``.
+        ignore_self : bool
+            If True, ignores the transition from one state to itself.
+            This is equivalent to setting the duration of all states to 1
+            sample.
+
+        Returns
+        -------
+        T : array of shape ``(n_cluster, n_cluster)``
+            Array of transition probability values from one label to another.
+            First axis indicates state ``"from"``. Second axis indicates state
+            ``"to"``.
+
+        Warnings
+        --------
+        When working with `~mne.Epochs`, this method will take into account
+        transitions that occur between epochs. This could lead to wrong
+        interpretation when working with discontinuous data.
+        To avoid this behaviour, make sure to set the ``reject_edges``
+        parameter to ``True`` when predicting the segmentation.
+        """  # noqa: E501
+        _check_value(
+            stat,
+            (
+                "count",
+                "probability",
+                "proportion",
+                "percent",
+            ),
+            "stat",
+        )
+        n_clusters = len(self._cluster_centers_)
+        T = _BaseSegmentation._compute_transition_matrix(
+            self._labels,
+            n_clusters=n_clusters,
+            stat=stat,
+            ignore_self=ignore_self,
+        )
+        return T
+
+    @staticmethod
+    def _compute_transition_matrix(
+        labels, n_clusters, ignore_self=True, stat="probability"
+    ):
+        """Compute observed transition."""
+        # reshape if epochs
+        labels = labels.reshape(-1)
+        # ignore transition to itself (i.e. 1 -> 1)
+        if ignore_self:
+            labels = [s for s, _ in itertools.groupby(labels)]
+
+        T = np.zeros(shape=(n_clusters, n_clusters))
+        # number of transitions
+        for i, j in zip(labels, labels[1:]):
+            if i == -1 or j == -1:
+                continue  # ignore unlabeled
+            T[i, j] += 1
+
+        # transform to probability
+        if stat != "count":
+            T = T / T.sum(axis=1, keepdims=True)
```

Comment:
For now the tests are failing because of this change since you included a test that covers this case.
Alternatively, to detecting and handling the issue at the beginning, let's at least use a vectorized version of the oop:

```
with np.errstate(divide='ignore'):
    T = T / T.sum(axis=1, keepdims=True)
    np.nan_to_num(T, posinf=0, copy=False)
```

Reply:
Ok, let's keep the handing at the end then. I was also thinking about the opposite case where you have more labels than `n_clusters` (which could have been handled together at the beginning of the function), but I guess this case should not occur without tempering into the private variables of a segmentation. 

I fixed the test with `np.errstate(divide='ignore', invalid="ignore"):`, because 5 / 0 yields inf while 0 / 0 yields nan in numpy, and that's 2 different errors.